### PR TITLE
Feature user created

### PR DESCRIPTION
**Pull request by YahyaDari**
I am sending files for review.

### CODE REVIEW

**Review by najlaalhaj — CHANGES_REQUESTED** [review]
please add a  List<Project> property with its getter and setter methods, will be great if you call it (projects). will need it for the user story we are working on.

**Comment by YahyaDari** [comment]
Oky Najla ,i'll do that .

**Review by Jabbar-Hussain — APPROVED** [review]

**Review by najlaalhaj — APPROVED** [review]
Good Job!

**Comment by YahyaDari** [comment]
Hi Najla. that was not my task this sprint ,but i'm going to do that next sprint:)

File: src/main/java/io/orten/nano/business/UserAPI.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+
+@Path("/")
+public class UserAPI {
+
+    public static List<User> u_list = new ArrayList<User>();
```
Comment (munHunger) [inline src/main/java/io/orten/nano/business/UserAPI.java]: Switch to using camel case and give it a more meaningful name.
i.e. userList or similar

File: src/main/java/io/orten/nano/business/UserAPI.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @GET
+    @Path("/get/user/{userID}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getUser(@PathParam("userID") long userID) {
+        try {
+            User user = UserService.getUser(userID);
+            return Response.status(200).entity(user).build();
```
Comment (munHunger) [inline src/main/java/io/orten/nano/business/UserAPI.java]: Switch from using static integers to HttpServletResponse.SC_OK

File: src/main/java/io/orten/nano/impl/UserService.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserService {
+
+        public static List<User> u_list = new ArrayList<User>();
```
Comment (munHunger) [inline src/main/java/io/orten/nano/impl/UserService.java]: Give it a more meaningful camelCased name

File: src/main/java/io/orten/nano/impl/UserService.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        public static User getUser(long userID) throws Exception {
+            Session session = Database.getSession();
+            User user = session.get(User.class, userID);
```
Comment (munHunger) [inline src/main/java/io/orten/nano/impl/UserService.java]: If this throws an exception then the session isn't closed and you have a resource leak.
This should probably be in a try-finally block

File: src/main/java/io/orten/nano/impl/UserService.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            Transaction tx = null;
+            try {
+                tx = session.beginTransaction();
+                session.save(user);
+                tx.commit();
+            } catch (HibernateException e) {
+                if (tx != null) tx.rollback();
+                throw e;
+            } finally {
+                session.close();
```
Comment (munHunger) [inline src/main/java/io/orten/nano/impl/UserService.java]: Well done on the finally block. just make everything look like this :)

File: src/main/java/io/orten/nano/model/User.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import javax.persistence.Table;
+import java.util.Date;
+
+@Entity
+@Table(name = "user")
+public class User {
+
+    @Id
+    @Column(name = "userID")
+    private long userID;
```
Comment (munHunger) [inline src/main/java/io/orten/nano/model/User.java:15]: Does it work having these as private?
Might prove problematic when parsing to and from JSON

File: src/main/java/io/orten/nano/impl/UserService.java (hunk 5 above)
Comment (YahyaDari) [reply src/main/java/io/orten/nano/impl/UserService.java]: 👍 :)

File: src/main/java/io/orten/nano/business/UserAPI.java (hunk 1 above)
Comment (YahyaDari) [reply src/main/java/io/orten/nano/business/UserAPI.java]: Replaced u_list by userList

File: src/main/java/io/orten/nano/impl/UserService.java (hunk 3 above)
Comment (YahyaDari) [reply src/main/java/io/orten/nano/impl/UserService.java]: Replaced u_list by userList

File: src/main/java/io/orten/nano/business/UserAPI.java (hunk 2 above)
Comment (YahyaDari) [reply src/main/java/io/orten/nano/business/UserAPI.java]: Replaced 200 by HttpServletResponse.SC_OK 
and 
Replaced 500 by HttpServletResponse.SC_INTERNAL_SERVER_ERROR

File: src/main/java/io/orten/nano/impl/UserService.java (hunk 4 above)
Comment (YahyaDari) [reply src/main/java/io/orten/nano/impl/UserService.java]: I have fixed it

File: src/main/java/io/orten/nano/model/User.java (hunk 6 above)
Comment (YahyaDari) [reply src/main/java/io/orten/nano/model/User.java:15]: Its working

**Pull request merged**